11-lorawan/test_spec11.py

```diff
@@ -0,0 +1,118 @@
+import time
+import pytest
+from testutils.shell import GNRCLoRaWANSend, ifconfig, lorawan_netif
+from riotctrl_shell.netif import Ifconfig
+import base64
+import json
+
+APP = 'examples/gnrc_lorawan'
+pytestmark = pytest.mark.rc_only()
+
+APP_PAYLOAD = "This is RIOT!"
+DOWNLINK_PAYLOAD = "VGhpcyBpcyBSSU9U"
+APP_PORT = 2
+RX2_DR = 3
+
+
+class Shell(Ifconfig, GNRCLoRaWANSend):
+    pass
+
+
+def run_lw_test(node, client, userdata, iface, app_id, dev_id):
+    # Disable confirmable messages
+    node.ifconfig_flag(iface, "ack_req", enable=False)
+
+    # Push a downlink message to the TTN server
+    dl_data = {"payload_raw": DOWNLINK_PAYLOAD, "port": APP_PORT,
+               "confirmed": True}
+
+    client.publish("{}/devices/{}/down".format(app_id, dev_id),
+                   json.dumps(dl_data))
+
+    # Send a message. The send function will return True if the downlink is
+    # receives (as expected)
+    assert node.send(iface, APP_PAYLOAD) is True
+    time.sleep(3)
+
+    message = base64.b64decode(userdata["msg"]["payload_raw"])
+    assert message.decode('ascii') == APP_PAYLOAD
```

Comment:
And this could be something like

```py
def last_received_payload(self):
    message = base64.b64decode(userdata["msg"]["payload_raw"])
    return message.decode('ascii')
```

(unrelated but important: are you 100% sure this message is _always_ ASCII encoded?)

Reply:
if it's not ASCII it means the downlink failed (wrong keys, etc). But maybe I can try to add a proper error message